File: 153.寻找旋转排序数组中的最小值.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int l = 0, h = nums.size() - 1;
        while(l < h){
            int mid = l + (h - l) / 2;
            if(nums[mid] < nums[h])
                h = mid;
            else
                l = mid + 1;
        }
        return nums[l];
    }
};

File: 二分查找/153.寻找旋转排序数组中的最小值.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var findMin = function(nums) {
    let [l, r] = [0, nums.length - 1];
    while (l < r) {
        let mid = Math.floor((r - l) / 2) + l;
        if (nums[mid] < nums[r]) { // 右半部分是有序的
            r = mid;
        } else { // 左半部分是有序的
            l = mid + 1;
        }
    }
    return nums[l];
};